### PR TITLE
Drop python 3.9, Add creation functions.

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This PR:

- Drops Python 3.9 support, as `sparse` already did it.
- Removes:
  - `poetry.lock` file. I've read that:
    https://stackoverflow.com/a/61076546/6717327
    > The official recommendation of the poetry maintainers is to commit the lockfile if you develop a deployable application (as opposed to a library).
  - and `requirements.txt` (unused)
- Adds a few methods to `Tensor`: `>`, `<`, `==`, etc...
- Adds `bitwise_*` aliases (already tested underlying `Tensor` methods)
- Introduces `full`, `zeros` and `ones` creation functions (We need to fix https://github.com/willow-ahrens/Finch.jl/pull/512 first, as COO is the format used by these functions)

### CODE REVIEW

File: tests/test_sparse.py [hunk 1]
```diff
@@ -175,6 +175,25 @@ def test_asarray(arr2d, arr3d, order, format):
         assert_equal(result.todense(), arr)
 
 
+@pytest.mark.parametrize("shape", [10, (3, 3), (2, 1, 5)])
+@pytest.mark.parametrize("dtype", [None, finch.int64, finch.float64])
+def test_full_ones_zeros(shape, dtype):
+    res = finch.full(shape, 2.0, dtype=dtype)
+    assert_equal(res.todense(), np.full(shape, 2.0, dtype))
+    res = finch.full_like(res, 3.0, dtype=dtype)
+    assert_equal(res.todense(), np.full(shape, 3.0, dtype))
```
Comment (hameerabbasi) [inline tests/test_sparse.py]: We should test all these functions without `dtype` as well.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Thanks for working on this, @mtsokol!

File: src/finch/tensor.py [hunk 2]
```diff
@@ -515,6 +543,39 @@ def asarray(obj, /, *, dtype=None, format=None):
...
+    shape: int | tuple[int, ...],
+    fill_value: jl_dtypes.number,
+    *,
+    dtype: DType | None = None,
+) -> Tensor:
+    if isinstance(shape, int):
+        shape = (shape,)
+    if dtype is not None:
+        fill_value = dtype(fill_value)
+    return Tensor(jl.Tensor(jl.SparseCOO[len(shape)](jl.Element(fill_value), shape)))
```
Comment (willow-ahrens) [inline src/finch/tensor.py]: pass in shape as an argument to `jl.Tensor` to initialize it.

**Review by willow-ahrens — CHANGES_REQUESTED** [review]
See my comments regarding segfaults. Also, do we want to add a parameter to specify a format to any of these constructors?

In Julia, there's a distinction between `zeros` (dense) and `spzeros` (sparse CSR). I wonder if we could add a `format` flag, or maybe a `issparse` boolean?

File: tests/test_sparse.py (hunk 1 above)
Comment (mtsokol) [reply tests/test_sparse.py]: Thanks for pointing out! Now one of the `dtype` param values is `None` which is equivalent to not passing it.

File: src/finch/tensor.py (hunk 2 above)
Comment (mtsokol) [reply src/finch/tensor.py]: Done!

**Comment by mtsokol** [comment]
> See my comments regarding segfaults. Also, do we want to add a parameter to specify a format to any of these constructors?

Thank you! Now it works as expected.

> In Julia, there's a distinction between `zeros` (dense) and `spzeros` (sparse CSR). I wonder if we could add a `format` flag, or maybe a `issparse` boolean?

Sure! Let me add `format` keyword arg that will accept `dense` and `coo` for now.

**Comment by willow-ahrens** [comment]
Eventually, maybe we can use Hameer's PR for all of these format arguments

File: src/finch/tensor.py [hunk 3]
```diff
@@ -809,6 +883,30 @@ def atan2(x: Tensor, other: Tensor, /) -> Tensor:
...
+def bitwise_right_shift(x1: Tensor, x2: Tensor, /) -> Tensor:
+    return x1 >> x2
+
+
+def bitwise_xor(x1: Tensor, x2: Tensor, /) -> Tensor:
+    return x1 ^ x2
+
+
+def bitwise_invert(x: Tensor, /) -> Tensor:
+    return ~x
```
Comment (hameerabbasi) [inline src/finch/tensor.py]: Can we replace these with `from operator import ... [as ...]`?

Comment (hameerabbasi) [reply src/finch/tensor.py]: While we're at it, let's do it for all functions where this may apply.

Comment (mtsokol) [reply src/finch/tensor.py]: Done! (it reduced a bit of boilerplate code)

**Review by hameerabbasi — APPROVED** [review]

File: src/finch/tensor.py [hunk 4]
```diff
@@ -515,6 +543,52 @@ def asarray(obj, /, *, dtype=None, format=None):
...
+    format: str = "coo",
+) -> Tensor:
+    if isinstance(shape, int):
+        shape = (shape,)
+    if dtype is not None:
+        fill_value = dtype(fill_value)
+
+    if format == "coo":
+        return Tensor(jl.Tensor(jl.SparseCOO[len(shape)](jl.Element(fill_value)), *shape))
+    if format == "dense":
```
Comment (willow-ahrens) [inline src/finch/tensor.py:560]: LGTM! I'll add an issue to update these.

**Review by willow-ahrens — APPROVED** [review]
LGTM!

**Review by willow-ahrens — COMMENTED** [review]
Oh! Also, I just want to say that you can use the Finch.FinchNotation.and and Finch.FinchNotation.or for logical and/or (the operations which return 1 or 0 as opposed to bitwise). Julia doesn't usually name these as functions

**Pull request merged**